File: Python/Software Development Module - Python/Revision/Exam2016/LuasTicket/LuasTicketApp.py
"""
LuasTicketApp.py
Billy Cussen
31/10/2020
"""

#Object
import gc
import LuasTicket
from LuasTicket import myLuasTicket
import CustomerInfo
from CustomerInfo import myCustomerInfo

#Variables
i = 0
repeat = "y"
ticketHistory = []

while repeat == "y":

    #Input - Ticket Type
    inputCheck = True
    while inputCheck:
        try:
            ticketType = input(str("Please enter Ticket type (\"Adult\" or \"Child\"): "))
            ticketType = ticketType.lower()
            if ticketType != "adult" and ticketType != "child":
                raise Exception
        except Exception:
            print("You need to enter \"Adult\" or \"Child\"")
        else:
            inputCheck = False

    while inputCheck == False:
        try:
            numberOfZones = input("Please enter the number of Zones you will visit: ")
            numberOfZones = int(numberOfZones)
            if numberOfZones <= 0 or numberOfZones > 8:
                raise Exception
        except Exception:
            print("You need to enter a number between 1 and 8")
        else:
            inputCheck = True

    #Process & Output
    myCustomerInfoCurrent = myCustomerInfo(ticketType,numberOfZones)
    myLuasTicketApp = myLuasTicket(myCustomerInfoCurrent)
    try:
        cost = myLuasTicketApp.getLuasTicket()
        print("Your Luas Ticket will cost: "+str(cost))
        i = i+1
        ticketHistory.append("Ticket Price "+str(i)+": "+str(cost))
    except Exception as e:
        print(e)
    else:
        myCustomerInfoCurrent = None
        myLuasTicketApp = None
        gc.collect

    #Input - Repeat?
    repeat = ""
    while repeat != "y" and repeat != "n":
        repeat = input(str("Please enter \"y\" to retry, enter \"n\" to exit: "))
        repeat = repeat.lower()
        if repeat != "y" and repeat != "n":
            print("You need to enter \"y\" or \"n\"")

#Write History to Text File
file1 = open("LuasTicketApp.txt","r+") 
file1.truncate(0)

file1.write("***LUAS TICKET APP HISTORY***\n")
for ticket in ticketHistory:    
    file1.write(ticket+"\n")
file1.close()
file1 = None

#Read Contents and print Results
with open("LuasTicketApp.txt") as text_file:
    contents = text_file.read()
print(contents)

gc.collect()

print("Thanks for using my Luas Ticket App!")

File: Java/Software Development Module - Java/Revision/Exam2016/LuasTicketApp.java
/*
LuasTicketApp.java
Billy Cussen
14/08/2020
*/

import java.util.InputMismatchException;
import java.util.ArrayList;
import java.util.Scanner;

public class LuasTicketApp{

    public static void main(String [] args){

        //Variables
        String repeat = new String();
        String result;
        double total = 0.0;
        boolean inputCheck;

        //Objects
        ArrayList<Double> ticketHistory = new ArrayList<Double>();
        Scanner keyboard;
        CustomerInfo myCustomerInfo; 
        LuasTicket myLuasTicket;

        do{
            myCustomerInfo = new CustomerInfo(); 

            //Input
            do{
                keyboard = new Scanner(System.in);
                inputCheck = true;

                try{
                    System.out.print("Please enter ticket Type (\"Adult\" or \"Child\"): ");
                    myCustomerInfo.setPerson(keyboard.nextLine());

                    if(!(myCustomerInfo.getPerson().equalsIgnoreCase("Adult")||myCustomerInfo.getPerson().equalsIgnoreCase("Child"))){
                        throw new InputMismatchException("You need to enter either \"Adult\" or \"Child\"!");
                    }

                }catch(InputMismatchException e){
                    System.out.println(e.getMessage());
                    inputCheck = false;
                } finally{
                    keyboard = null;
                }

            }while(!inputCheck);

            do{
                keyboard = new Scanner(System.in);
                inputCheck = true;

                try{
                    System.out.print("Please enter the number of Zones you will Travel through: ");
                    myCustomerInfo.setZones(keyboard.nextInt());

                    if(myCustomerInfo.getZones() <= 0 || myCustomerInfo.getZones() > 8){
                        throw new NumberFormatException();
                    }

                }catch(InputMismatchException | NumberFormatException e){
                    System.out.println("You can travel to between 1 and 8 Zones, please enter a number!");
                    inputCheck = false;
                } finally{
                    keyboard = null;
                }

            }while(!inputCheck);

            //Process
            myLuasTicket = new LuasTicket(myCustomerInfo);
            myCustomerInfo = null;
            result = "The Cost of your Ticket is: "+myLuasTicket.getLuasTicket();
            ticketHistory.add(myLuasTicket.getLuasTicket());
            myLuasTicket = null;

            //Output
            System.out.println(result);

            do{
                keyboard = new Scanner(System.in);
                System.out.print("Enter \"Y\" or \"N\" if you would or wouldn't like another ticket: ");
                repeat = keyboard.nextLine();

                if(!(repeat.equalsIgnoreCase("Y")||repeat.equalsIgnoreCase("N"))){
                    System.out.println("You need to enter \"Y\" or \"N\"!");
                }

            }while(!(repeat.equalsIgnoreCase("Y")||repeat.equalsIgnoreCase("N")));

        }while(repeat.equalsIgnoreCase("Y"));

        for(double e: ticketHistory){
            total += e;
        }

        ticketHistory = null;
        System.out.println("The Total for all of your Tickets is: "+total);
    }
}